packages/auth/lib/Users.js

```diff
@@ -406,18 +412,23 @@ const upsertUserAndConsents = async ({ pgdb, email, consents, req }) => {
     consents
   })
 
-  const user = existingUser ||
-    await pgdb.public.users.insertAndGet({
+  const userFields = await ensureRequiredFields({
+    user: existingUser,
+    email,
+    providedFields: fields,
+    pgdb
+  })
+
+  const user = !existingUser
+    ? await pgdb.public.users.insertAndGet({
+      ...userFields,
       email,
       verified: true
     })
-  if (!user.verified) {
-    await pgdb.public.users.updateOne({
-      id: user.id
-    }, {
+    : await pgdb.public.users.updateAndGetOne({ id: existingUser.id }, {
```

Comment:
I'd avoid always updating the user object on every `authorizeSession`. Check if userFields actually contains something. Also missing `updatedAt: new Date()`.